command/snapshot/inspect/snapshot_inspect.go

```diff
@@ -56,27 +65,181 @@ func (c *cmd) Run(args []string) int {
 	}
 	defer f.Close()
 
-	meta, err := snapshot.Verify(f)
+	readFile, meta, err := snapshot.Read(nil, f)
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Error reading snapshot: %s", err))
+	}
+	stats, offset, err := enhance(readFile)
 	if err != nil {
-		c.UI.Error(fmt.Sprintf("Error verifying snapshot: %s", err))
+		c.UI.Error(fmt.Sprintf("Error extracting snapshot data: %s", err))
 		return 1
 	}
+	// Outputs the original style of inspect information
+	legacy, err := c.legacyStats(meta)
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Error outputting snapshot data: %s", err))
+	}
+	c.UI.Info(legacy.String())
+
+	// Outputs the more detailed snapshot information
+	enhanced, err := c.readStats(stats, offset)
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Error reading snapshot stats: %s", err))
+		return 1
+	}
+	c.UI.Info(enhanced.String())
+
+	return 0
+}
 
+// legacyStats outputs the expected stats from the original snapshot
+// inspect command
+func (c *cmd) legacyStats(meta *raft.SnapshotMeta) (bytes.Buffer, error) {
 	var b bytes.Buffer
 	tw := tabwriter.NewWriter(&b, 0, 2, 6, ' ', 0)
 	fmt.Fprintf(tw, "ID\t%s\n", meta.ID)
 	fmt.Fprintf(tw, "Size\t%d\n", meta.Size)
 	fmt.Fprintf(tw, "Index\t%d\n", meta.Index)
 	fmt.Fprintf(tw, "Term\t%d\n", meta.Term)
 	fmt.Fprintf(tw, "Version\t%d\n", meta.Version)
-	if err = tw.Flush(); err != nil {
+	if err := tw.Flush(); err != nil {
 		c.UI.Error(fmt.Sprintf("Error rendering snapshot info: %s", err))
```

Comment:
minor: there's a `UI.Error` print in the caller, so if this error is hit we'll get two error lines instead of 1. We can remove either this one, or the one on line 80 above.